File: plugin-dotnet-common/src/main/kotlin/jetbrains/buildServer/dotnet/Verbosity.kt


package jetbrains.buildServer.dotnet

enum class Verbosity(val id: String, val description: String) {
    Quiet("Quiet", "Quiet"),
    Minimal("Minimal", "Minimal"),
    Normal("Normal", "Normal"),
    Detailed("Detailed", "Detailed"),
    Diagnostic("Diagnostic", "Diagnostic");

    companion object {
        fun tryParse(id: String): Verbosity? {
            return Verbosity.values().singleOrNull { it.id.equals(id, true) }
        }
    }
}

File: plugin-dotnet-agent/src/main/csharp/TeamCity.Dotnet.Plugin.Agent/TeamCity.Dotnet.Plugin.Agent.AssemblyLevelTestFilter/Infrastructure/CommandLine/Commands/Verbosity.cs
namespace TeamCity.Dotnet.Plugin.Agent.AssemblyLevelTestFilter.Infrastructure.CommandLine.Commands;

internal enum Verbosity
{
    Quiet,      // no logs
    Minimal,    // errors only
    Normal,     // default (information logs)
    Detailed,   // 
    Diagnostic  // debug
}